src/TropicalGeometry/groebner_basis.jl

```diff
@@ -307,7 +307,7 @@ function desimulate_valuation(w::Vector{QQFieldElem}, nu::TropicalSemiringMap{K,
     end
     return w[2:end]
 end
-function desimulate_valuation(w::Vector{QQFieldElem}, nu::TropicalSemiringMap{K,p,typeof(max)}; perturbation::Union{Nothing,Vector}=nothing) where {K,p}
+function desimulate_valuation(w::AbstractVector{QQFieldElem}, ::TropicalSemiringMap{K,p,typeof(max)}; perturbation::Union{Nothing,AbstractVector}=nothing) where {K,p}
     @req w[1]<0 "invalid weight vector"
     # scale the vector so that first entry is -1, then remove first entry
     w ./= -w[1]
```

Comment:
```suggestion
    w = w ./ -w[1]
```
Or even better I guess directly `w = w[2:end] ./ -w[1]`, but I cannot change some of the lines below in this suggestion to remove the `[2:end]`.

This does not modify the original vector unlike the broadcasted assignment. (And doesn't require `setindex!`)